### PR TITLE
Update the SDK and Version dropdowns to include location hash

### PR DESCRIPTION
**Pull request by stnguyen90**
## What does this PR do?

Update the SDK and Version dropdowns to include location hash. Prior to this, if you:

1. Browse to API docs like the Databases API docs
2. Click on a link on the right side like Get Document
3. Switch your SDK

You would jump to the top of the page instead of still being on the Get Document API.

https://user-images.githubusercontent.com/1477010/197569767-39550e5c-679b-48a2-959f-c87274d815e9.mov

After this, you would stay on the API you last clicked on after switching the version or SDK.

## Test Plan

Manual

## Related PRs and Issues

Based off the following PR:

* https://github.com/appwrite/docs/pull/269

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

**Comment by gewenyu99** [comment]
This does preserve location for the most part. When we change the name of endpoints, it'll jump back to the top of the page because there's no matching hash, which is fine.

It works really well when I switch SDKs, which is great.

Now the real question is, since we're already adding JS, who don't we track scroll location?

**Comment by stnguyen90** [comment]
> Now the real question is, since we're already adding JS, who don't we track scroll location?

@gewenyu99, that seems much higher effort 😅

**Review by gewenyu99 — APPROVED** [review]
I tested this on homepage, works great. Idk if the implementation is optional, I'll leave it to engineering to decide :)

**Review by PineappleIOnic — APPROVED** [review]

**Review by gewenyu99 — APPROVED** [review]
Tested again, lgtm

**Comment by eldadfux** [comment]
Would love to get some more context are a short example to fully understand the new suggested behavior.

**Comment by stnguyen90** [comment]
@eldadfux, I've updated the description.

**Comment by gewenyu99** [comment]
@stnguyen90 The scroll bars look so weird there! is this what it looks like on windows?

File: app/views/docs/service.phtml [hunk 1]
```diff
@@ -163,7 +163,7 @@ $paramSDK = (!empty($sdk)) ? '?sdk='.$sdk : '';
                                     : ['Default' => 'examples'];
                                 ?>
                                 <?php foreach ($docDirectories as $docDirectoryKey => $docDirectory): ?>
-                                <option value="/docs/<?php echo $this->escape($family); ?>/<?php echo $serviceName; ?>?sdk=<?php echo $this->escape($key); ?>-<?php echo $this->escape(strtolower($docDirectoryKey)); ?>"<?php echo (($key . '-' . strtolower($docDirectoryKey)) === $sdk) ? ' selected' : ''; ?>>
+                                <option data-ls-attrs="value=/docs/<?php echo $this->escape($family); ?>/<?php echo $serviceName; ?>?sdk=<?php echo $this->escape($key); ?>-<?php echo $this->escape(strtolower($docDirectoryKey)); ?>{{router.hash}}" <?php echo (($key . '-' . strtolower($docDirectoryKey)) === $sdk) ? ' selected' : ''; ?>>
```
Comment (eldadfux) [inline app/views/docs/service.phtml:166]: @stnguyen90 please validate this value is properly escaped and don't create an XSS injection by mistake.

File: app/views/docs/service.phtml [hunk 2]
```diff
@@ -174,7 +174,7 @@ $paramSDK = (!empty($sdk)) ? '?sdk='.$sdk : '';
             </div>
             <div class="col span-4 clear">
                 <form class="pull-end desktops-only" onchange="this.submit()"
-                      method="get" action="/docs/<?php echo $this->escape($family); ?>/<?php echo $serviceName; ?>">
+                      method="get" data-ls-attrs="action=/docs/<?php echo $this->escape($family); ?>/<?php echo $serviceName; ?>{{router.hash}}">
```
Comment (eldadfux) [inline app/views/docs/service.phtml:177]: Same about the XSS injection.

Comment (stnguyen90) [reply app/views/docs/service.phtml:177]: Seems like the hash is sanitized a bit such that double quotes are converted to `%22` so you wouldn't be able to add an additional HTML attribute to execute JS:

image

**Pull request merged**